Allowing dumping link with filter mask

cathay4t opened the pull request:
The linux bridge vlan filter information is only available when
`RTEXT_FILTER_BRVLAN` set in `Nla::ExtMask`.

This patch introduce `LinkGetRequest::set_filter_mask()` method allowing
such usage via:

    let (connection, handle, _) = rtnetlink::new_connection().unwrap();
    tokio::spawn(connection);
    let mut links = handle.link().get().set_filter_mask(RTEXT_FILTER_BRVLAN)
        .execute();

The similar libnetlink function is `rtnl_linkdump_req_filter()`.

Also changed these constants to u32 in compliance of libnetlink and
kernel codes:

 * `RTEXT_FILTER_VF`
 * `RTEXT_FILTER_BRVLAN`
 * `RTEXT_FILTER_BRVLAN_COMPRESSED`
 * `RTEXT_FILTER_SKIP_STATS`

Example code for query linux bridge vlan information is included in
`rtnetlink/examples/get_links.rs` as `dump_bridge_filter_info()`.

little-dude commented on rtnetlink/src/link/get.rs:
```diff
@@ -34,6 +34,17 @@ impl LinkGetRequest {
     }
 
+    /// Setting filter mask(e.g. RTEXT_FILTER_BRVLAN and etc)
+    pub fn set_filter_mask(mut self, family: u8, filt_mask: u32) -> Self {
```
This is kind a bit redundant with `new()`, could we make this simply:

```rust
pub fn set_filter_mask(mut self, family: u8, filt_mask: u32) -> Self {
     self.message.header.interface_family = family;
     self.message.nlas.push(Nla::ExtMask(filter_mask));
     self
}
```

Then it would be called like:

```rust
LinkGetRequest::new(handle)
    .set_filter_mask(family, filt_mask)
    .execute()
```

Also, do we necessarily need the address family when setting a filter?
Or should this be a different `set_interface_family()` method?

cathay4t replied:
Changing the `LinkGetRequest::new()` would break the API which I think we can discuss that in another pull request while this PR focusing on extending existing API for `filter mask`.
But if you prefer to do it in a bundle, I am OK with it, just let me know.

cathay4t replied:
I tested several days ago, it is required for vlan filtering, but let me test again.

cathay4t replied:
It it required to set `interface_family`. Without it, the `AfBridgeSpec` will not show. Tested on kernel 5.4.43(archlinux).

For kernel code, the `validate_linkmsg()` in `net/core/rtnetlink.c` will ensure the returned AfSpec interface family matches, and `br_fill_ifinfo()` in `net/bridge/br_netlink.c` set the `hdr->ifi_family = AF_BRIDGE`.

little-dude replied:
Sounds good, thank you.

JohnTitor commented on rtnetlink/Cargo.toml:
```diff
@@ -17,6 +17,7 @@ log = "0.4.8"
 thiserror = "1"
 netlink-packet-route = { path = "../netlink-packet-route", version = "0.2" }
 netlink-proto = { path = "../netlink-proto", version = "0.2" }
+netlink-sys = {path = "../netlink-sys", version = "0.2" }
```
This should be dev-dep.

cathay4t commented:
New patch uploaded.

Changes since last one:
 * Changed to in place update without `new()`.
 * Fixed the `rtnetlink/Cargo.toml` to include the new dependency as dev only.

little-dude commented:
@cathay4t do you want me to release new crates with all your recent changes, or do you have some more you might push?

cathay4t commented:
@little-dude I don't have work in progress[0]. Release a new crates could helps. So that my [project][1] could depend on.

[1]: https://github.com/nispor/nispor

little-dude commented:
Cool. Will do tonight!

cathay4t commented:
@little-dude Any plan for the new release?

little-dude commented:
Sorry I forgot. I just released these new versions:

- audit 0.2.1
- rtnetlink 0.3.0
- netlink-packet-audit 0.2.0
- netlink-proto 0.4.0 (should have been 0.3.0 but I messed up)
- netlink-packet-route 0.3.0
- netlink-packet-core 0.2.0
- netlink-packet-utils 0.2.0
- netlink-sys 0.3.0
